client/v2/client.go

```diff
@@ -0,0 +1,371 @@
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/influxdb/influxdb/models"
+)
+
+type Config struct {
+	// URL of the InfluxDB database
+	URL *url.URL
+
+	// Username is the influxdb username, optional
+	Username string
+
+	// Password is the influxdb password, optional
+	Password string
+
+	// UserAgent is the http User Agent, defaults to "InfluxDBClient"
+	UserAgent string
```

Comment:
I guess people might like to set the agent to any client program they build. Makes sense.